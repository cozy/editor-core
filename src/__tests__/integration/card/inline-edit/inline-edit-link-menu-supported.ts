import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  getDocFromElement,
  editable,
  copyToClipboard,
  gotoEditor,
  manuallyEmptyLinkToolbar,
  linkUrlSelector,
} from '../../_helpers';

import { waitForInlineCardSelection } from '@atlaskit/media-integration-test-helpers';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

BrowserTestCase(
  `card: selecting an inline card and choosing a new page from edit-link menu should update title and url for supported link`,
  {
    skip: ['edge'],
  },
  async (client: ConstructorParameters<typeof Page>[0], testName: string) => {
    const page = new Page(client);

    // Copy stuff to clipboard
    await copyToClipboard(page, 'https://www.atlassian.com');

    await gotoEditor(page);
    // Paste the link
    await page.paste();

    await waitForInlineCardSelection(page);
    await page.click('button[aria-label="Edit link"]');

    // Links chosen from edit-link menu require both URL and label to have changed
    // to resolve as Smartlinks
    await manuallyEmptyLinkToolbar(page);

    // Choosing a supported link
    await page.type(linkUrlSelector, 'home opt-in');
    await page.keys(['ArrowDown', 'Return']);

    await waitForInlineCardSelection(page);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
