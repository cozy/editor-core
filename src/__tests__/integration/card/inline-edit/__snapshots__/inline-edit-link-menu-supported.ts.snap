// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card: selecting an inline card and choosing a new page from edit-link menu should update title and url for supported link 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "data": Object {
              "@context": "https://www.w3.org/ns/activitystreams",
              "@type": "Document",
              "name": "https://product-fabric.atlassian.net/wiki/display/H/Home+opt-in+requests",
              "url": "https://product-fabric.atlassian.net/wiki/display/H/Home+opt-in+requests",
            },
            "url": null,
          },
          "type": "inlineCard",
        },
        Object {
          "text": "  ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
